execution/src/execution.rs

```diff
@@ -269,6 +269,83 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
         }
         Ok(logs)
     }
+
+       pub async fn get_filter_changes(
+        &self, 
+        filter_id: &U256,
+    ) -> Result<Vec<Log>> {
+        let logs = self.rpc.get_filter_changes(&filter_id).await?;
+        if logs.len() > MAX_SUPPORTED_LOGS_NUMBER {
+            return Err(
+                ExecutionError::TooManyLogsToProve(logs.len(), MAX_SUPPORTED_LOGS_NUMBER).into(),
+            );
+        }
+
+        for (_pos, log) in logs.iter().enumerate() {
```

Comment:
Can we dedupe this from the `get_logs` function? Maybe we can extract the duplicated code in a `verify_logs` function that takes the list of unverified logs as an input?

Reply:
Done!